### DISCUSSION TITLE
Question: Why is UTM taking up twice the memory limit I set in settings

### DISCUSSION THREAD

**Discussion by sainnhe** [utmapp/UTM#5647] [UTM Help (macOS)]
**Describe the issue**  
I set 6GB RAM limit in settings, but UTM actually costs 12GB. Is this expected behavior or a bug?

**Configuration**  
* UTM Version: 4.3.5
* macOS Version: 13.5.1
* Mac Chip (Intel, M1, ...): M1

**Comment by osy** [comment]
It's not a limit in settings, that's how much memory is visible to the host. It does not include any additional memory used for emulation. In this case a lot of memory is used for GPU acceleration.

**Comment by sainnhe** [comment]
@osy Thanks for your reply! I tried to switch to `virtio-ramfb` driver which doesn't support GPU acceleration, but there is no luck, UTM still eats tons of RAMs. Am I missing something?

Edit: By saying "tons of RAMs", I mean UTM will still use up to 12 GB RAMs while I only allocate 6 GB to this VM.

**Comment by sainnhe** [comment]
My configs:

- OS: Windows 11
- Arch: arm64
- Emulation: QEMU 7.2 ARM Virtual Machine
- Memory: 6144 MB
- CPU: Default
- CPU cores: 3
- JIT Cache: Default
- Adjustments:
  - UEFI
  - RNG
  - Ballon
  - TPM 2.0
  - Use virtualization
  - Use local clock
- Input: USB 3.0 XHCI
- USB sharing enabled
- Clipboard sharing enabled
- Directory sharing enabled (via SPICE WebDAV)
- GPU: virtio-ramfb
- Resize display to window size automatically
- Retina mode
